### PR TITLE
Add a new request-level field "cfhost"

### LINKED ISSUE #619: Add 'authority' to 'request'  in log structure
The field  representing 'authority' doesn't exist in logs.
Has to be added to  '_source.request' field

### PR DESCRIPTION
**Pull request by bartavelle**
This field is the "host" that is used to select the security policy.

Close #619 

Signed-off-by: Simon Marechal <bartavelle@gmail.com>

### CODE REVIEW

**Comment by bartavelle** [comment]
This modifies curielogger, review very much appreciated :)

File: curiefense/curielogger/pkg/entities/curie_proxy_log.go [hunk 1]
```diff
@@ -9,4 +9,5 @@ type CurieProxyLog struct {
 	Blocked     bool                   `json:"blocked"`
 	BlockReason map[string]interface{} `json:"block_reason"`
 	Tags        []string               `json:"tags"`
+	Cfhost      string                 `json:"cfhost"`
```
Comment (flaper87) [inline curiefense/curielogger/pkg/entities/curie_proxy_log.go:12]: Is this line really needed? Looking at the Rust changes it seems like we only need it in the `Request` struct.

Comment (bartavelle) [reply curiefense/curielogger/pkg/entities/curie_proxy_log.go:12]: TBH, I don't really know what I am doing ^^

Comment (bartavelle) [reply curiefense/curielogger/pkg/entities/curie_proxy_log.go]: Just remove that line.

Comment (bartavelle) [reply curiefense/curielogger/pkg/entities/curie_proxy_log.go]: Well, it didn't work :) I'll check taht monday

Comment (bartavelle) [reply curiefense/curielogger/pkg/entities/curie_proxy_log.go]: If I remove it, then this line won't work : https://github.com/curiefense/curiefense/pull/620/files#diff-eff0e05ed5e3dcc3b326158efb56f894ff52e568be3dff2017b85270036309f9R174

Am I doing something stupid here?

**Review by flaper87 — CHANGES_REQUESTED** [review]
needs rebase

**Comment by bartavelle** [comment]
Build is failing because `curieconfctl` is broken :(

**Comment by flaper87** [comment]
> Build is failing because `curieconfctl` is broken :(

Do we have a bug for this? Do you know already what's broken?

**Comment by bartavelle** [comment]
I tracked it down last time but just forgot, I think it was a dependency problem, lemme check it again.

**Comment by bartavelle** [comment]
I filed this issue : https://github.com/curiefense/curiefense/issues/645

**Comment by flaper87** [comment]
@bartavelle maybe we can rebase this PR now? Tests should pass

File: curiefense/curielogger/pkg/entities/curie_proxy_log.go (hunk 1 above)
Comment (xavier-rbz) [reply curiefense/curielogger/pkg/entities/curie_proxy_log.go:12]: After some debugging, I found out that the authority was already part of the received Json, we "just" needed to add it to the output structure. #657 does that.

**Comment by xavier-rbz** [comment]
With this PR, the authority would be transmitted twice from envoy to curielogger: in `attributes.authority`, and in the `cfhost` fields introduced by this PR. I believe the approach I follow in #657 avoids duplication.
I wish I had read this PR before debugging that, though :D

**Comment by bartavelle** [comment]
Then I guess we can close this PR?

**Comment by flaper87** [comment]
> Then I guess we can close this PR?

closing as it was superseded by #657

**Pull request closed without merging**